lib/inspec/schema/schema.rb

```diff
@@ -0,0 +1,64 @@
+require "json"
+require "lib/inspec/schema/primitives"
+require "lib/inspec/schema/exec_json"
+require "lib/inspec/schema/exec_json_min"
+require "lib/inspec/schema/profile_json"
+
+module Inspec
+  module OutputSchema
+    # Build our definitions
+    def self.build_definitions(schema_type)
+      # Init definition hash
+      definitions = {}
+
+      # Fetch all dependencies
+      defined_types = schema_type.all_depends
+
+      # Add them all to the definition hash
+      defined_types.each do |type|
+        definitions[type.name] = type.body
+      end
+
+      # Nest under "definitions" label
+      {
+        "definitions" => definitions,
+      }
+    end
+
+    # Helper function to automatically bundle a type with its dependencies
+    def self.finalize(schema_type)
+      schema_type.body.merge(OutputSchema.build_definitions(schema_type))
+    end
+
+    # using a proc here so we can lazy load it when we need
+    # TODO: Modernize
+    PLATFORMS = lambda do
+      require "train"
+      Train.create("mock").connection
+      Train::Platforms.export
+    end
+
+    LIST = {
+      "profile-json" => OutputSchema.finalize(Schema::ProfileJson::PROFILE),
+      "exec-json" => OutputSchema.finalize(Schema::ExecJson::OUTPUT),
+      "exec-jsonmin" => OutputSchema.finalize(Schema::ExecJsonMin::OUTPUT),
+      "platforms" => PLATFORMS,
```

Comment:
why isn't the proc here?

Reply:
Also, do we benefit from the lazy aspect at all? I can't tell when/where/how "pratforms" gets hit.

Reply:
Forgive me but what is the proc?

RE: Lazy aspect: Platforms is unchanged from the old code - I figured there must have been a reason for it at the time, but I can remove it now if necessary.